src/gwt/src/org/rstudio/studio/client/workbench/views/terminal/TerminalSession.java

```diff
@@ -211,6 +212,36 @@ protected void onDetach()
       super.onDetach();
       unregisterHandlers();
    }
+  
+   @Override
+   public void setVisible(boolean isVisible)
+   {
+      super.setVisible(isVisible);
+      if (isVisible)
+      {
+         // Inform the terminal that there may have been a resize. This could 
+         // happen on first display, or if the terminal was hidden behind other
+         // terminal sessions and there was a resize.
+         // A delay is needed to give the xterm.js implementation an
+         // opportunity to be ready for this.
+         
+         // TODO (gary) I already debounce heavily in XTermWidget.onResize, not
+         // sure why this additional level of delay is needed, but without it
+         // there are issues with xterm.js losing its mind when it is resized
+         // after re-emerging from behind other terminals. Why? Is this delay
+         // the best solution?
+         sendOnResizeLocal_.schedule(5);
```

Comment:
A couple of things to try:

1. Add some debug logging to tell you what the actual size of the element is here (i.e. just its `offsetHeight`/`offsetWidth`). The most common cause of this sort of behavior is elements trying to compute their size when they are either not visible or not attached to the DOM. 

2. In cases where you need to wait for the browser to finish a layout pass before picking up layout properties, the idiomatic thing to do is to use `Scheduler.get().scheduleDeferred()`, which schedules work to happen on the next event loop (something like a timer at 0ms).